### PR TITLE
Table component vanilla extract rewrite

### PR DESCRIPTION
**Pull request by bearni95**
## Reason:

Closes ["Recreate table component"](https://app.asana.com/0/1204649083736950/1204755978331326/f)

## Changes made (preferably with images/screenshots):
- recreated the table component, according to the latest figma design (included in the Asana ticket)
- added functionality to add a url to each `<tr>` row

## Check off the following:

- [x] I have reviewed my changes and run the appropriate tests.
- [ ] I have have run `rush change` to add the appropriate change logs.
- [x] I have added/edited docs.
- [x] I have added tutorials.
- [x] I have double checked and DEFINITELY added docs.

### CODE REVIEW

File: packages/libs/react-ui/src/components/Table/Table.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    rowCount: 3,
+    columnCount: 5,
+  },
+  render: ({ rowCount, columnCount }) => {
+    return (
+      <Table>
+        <Table.Head>
+          <Table.Tr>
+            {Array.from(Array(columnCount)).map((id, tdIdx) => {
+              return <Table.Tr.Th key={`td${tdIdx}`}>test {tdIdx}</Table.Tr.Th>;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Table/Table.stories.tsx:49]: I think keeping subcomponents at one level is best. So `Table.Th` should be sufficient

File: packages/libs/react-ui/src/components/Table/Table.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        return child;
+      })}
+    </table>
+  );
+};
+
+Table.Body = TBody;
+Table.Head = THead;
+Table.Tr = Tr;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Table/Table.tsx:38]: What is the purpose of this?

File: packages/libs/react-ui/src/components/Table/Table.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  children?: CompoundType<typeof TBody> | CompoundType<typeof THead>;
+}
+
+interface ITableComposition extends FC<ITable> {
+  Body: typeof TBody;
+  Head: typeof THead;
+  Tr: typeof Tr;
+}
+
+// eslint-disable-next-line react/prop-types
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Table/Table.tsx:19]: What is this reason for this? We want to avoid disabling lint errors when possible

**Review by eileenmguo — COMMENTED** [review]
Probably need to have a conversation with Isa about how link should work on this table

File: packages/libs/react-ui/src/components/Table/Table.tsx (hunk 2 above)
Comment (bearni95) [reply packages/libs/react-ui/src/components/Table/Table.tsx:38]: Was part of the original implementation for the table, simplifying it now

File: packages/libs/react-ui/src/components/Table/index.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+interface ITable extends FC<ITableProps> {
+  Body: FC<ITBodyProps>;
+  Head: FC<ITHeadProps>;
+  Tr: FC<ITrProps>;
+  Th: FC<IThProps>;
+  Td: FC<ITdProps>;
+}
+
+export const Table: ITable = TableContainer as ITable;
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/Table/index.ts:20]: Can you achieve this without a type cast?

**Review by eileenmguo — COMMENTED** [review]
Are the colors, active, and focus states correct? 
image
@isaozler @bearni95

File: packages/libs/react-ui/src/components/Table/index.ts (hunk 4 above)
Comment (bearni95) [reply packages/libs/react-ui/src/components/Table/index.ts]: As we discussed, replaced it with a container variable, exporting the `<Table>` component as `Table.Root`

File: packages/libs/react-ui/src/styles/sprinkles.css.ts [hunk 5]
```diff
@@ -13,6 +13,7 @@ export const breakpoints: Record<string, string> = {
 const systemProperties = defineProperties({
   properties: {
     fontFamily: vars.fonts,
+    borderSpacing: vars.sizes,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/styles/sprinkles.css.ts:16]: Looks like border-spacing is a table specific css property so it likely won't be used much. I would remove this from sprinkles

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**